Wshadow error in `function_ref.h`

**Describe your environment**

CentoOS

**Steps to reproduce**

Compile with clang

**What is the actual behavior?**


```
bazel-out/aarch64-fastbuild-clang-tidy/bin/external/io_opentelemetry_cpp/api/_virtual_includes/api/opentelemetry/nostd/function_ref.h:38:26: error: declaration shadows a field of 'function_ref<type-parameter-0-0 (type-parameter-0-1...)>' [-Werror,-Wshadow]
    invoker_  = [](void *callable_, Args... args) -> R {
                         ^
bazel-out/aarch64-fastbuild-clang-tidy/bin/external/io_opentelemetry_cpp/api/_virtual_includes/api/opentelemetry/nostd/function_ref.h:28:9: note: previous declaration is here
  void *callable_                = nullptr;
        ^
bazel-out/aarch64-fastbuild-clang-tidy/bin/external/io_opentelemetry_cpp/api/_virtual_includes/api/opentelemetry/nostd/function_ref.h:52:26: error: declaration shadows a field of 'function_ref<type-parameter-0-0 (type-parameter-0-1...)>' [-Werror,-Wshadow]
    invoker_  = [](void *callable_, Args... args) -> R {
                         ^
bazel-out/aarch64-fastbuild-clang-tidy/bin/external/io_opentelemetry_cpp/api/_virtual_includes/api/opentelemetry/nostd/function_ref.h:28:9: note: previous declaration is here
  void *callable_                = nullptr;
```

Clang is complaining about shadowing declaration in `function_ref.h`

--- a/api/include/opentelemetry/nostd/function_ref.h
+++ b/api/include/opentelemetry/nostd/function_ref.h
@@ -35,8 +35,8 @@ class function_ref<R(Args...)>
   void BindTo(F &f) noexcept
   {
     callable_ = static_cast<void *>(std::addressof(f));
-    invoker_  = [](void *callable_, Args... args) -> R {
-      return (*static_cast<FunctionPointer<F>>(callable_))(std::forward<Args>(args)...);
+    invoker_  = [](void *callable, Args... args) -> R {
+      return (*static_cast<FunctionPointer<F>>(callable))(std::forward<Args>(args)...);
     };
   }
 
@@ -49,8 +49,8 @@ class function_ref<R(Args...)>
       return BindTo(nullptr);
     }
     callable_ = reinterpret_cast<void *>(f);
-    invoker_  = [](void *callable_, Args... args) -> R {
-      return (F(callable_))(std::forward<Args>(args)...);
+    invoker_  = [](void *callable, Args... args) -> R {
+      return (F(callable))(std::forward<Args>(args)...);
     };
   }
 

